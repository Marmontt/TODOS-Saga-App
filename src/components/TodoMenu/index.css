.todo-menu-container {
    margin: 0 auto;
    width: 60vw;
    display: flex;
    align-items: center;
    flex-direction: row;
    border: 2px solid #e3e8eb;
    border-radius: 5px;
    background: white;
}

.todo-menu-input {
    height: 2em;
    width: 50%;
    padding: 5px;
    margin: 5px;
    border: 0;
    color: #7d7d7d;
    font-family: Roboto, serif;
    font-weight: 500;
    flex-grow: 1;
    font-size: 20px;
}
.todo-menu-input:focus{
    outline-width: 0;
}

.todo-menu-button {
    border-radius: 3px;
    width: 6em;
    height: 3em;
    color: #fff;
    background: #26a59a;
    border: 1px solid #26a59a;
    font-family: Roboto, serif;
    margin: 0 1em;
    font-weight: 500;
    outline: none;
}

.todo-menu-button:hover {
    cursor: pointer;
    background: #21877c;
}

.todo-menu-button:active {
    border: 1px solid #000;
    transform: translateY(2px);
}
