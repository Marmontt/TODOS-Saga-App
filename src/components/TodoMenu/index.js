import React, {useState} from 'react';
import {connect, useDispatch} from 'react-redux'

import * as todoActions from '../../state/todos/action'
import TodoColorBox from "./components/TodoColorBox";

import './index.css';

export const TodoMenu = (props) => {
    const [todoText, setTodoText] = useState('');
    const [chosenColor, setColor] = useState('#eb716a');
    const dispatch = useDispatch();

    return (
        <div className={'todo-menu-container'}>
            <input
                className={'todo-menu-input'}
                type="text"
                placeholder={"Add New Item"}
                value={todoText}
                onClick={() => setTodoText('')}
                onChange={e => setTodoText(e.target.value)}
            />
            <TodoColorBox chosenColor={chosenColor} setColor={setColor} color={'#eb716a'}/>
            <TodoColorBox chosenColor={chosenColor} setColor={setColor} color={'#fe74a2'}/>
            <TodoColorBox chosenColor={chosenColor} setColor={setColor} color={'#9170cc'}/>
            <TodoColorBox chosenColor={chosenColor} setColor={setColor} color={'#5cb3f9'}/>
            <TodoColorBox chosenColor={chosenColor} setColor={setColor} color={'#6ed3e5'}/>
            <TodoColorBox chosenColor={chosenColor} setColor={setColor} color={'#ffe07f'}/>
            <button onClick={() => dispatch(todoActions.addTodo(todoText, chosenColor))} className={'todo-menu-button'}>Add</button>
        </div>
    )
};


export default TodoMenu;
