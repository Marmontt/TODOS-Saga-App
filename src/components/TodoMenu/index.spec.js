import React from "react";
import { shallow } from "enzyme";

import TodoMenu from "./index";

describe("TodoMenu", () => {
  it("should render properly", () => {
    const wrapper = shallow(<TodoMenu />);
    expect(wrapper).toMatchSnapshot();
  });
});
