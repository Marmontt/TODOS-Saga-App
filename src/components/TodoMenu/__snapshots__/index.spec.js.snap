// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoMenu should render properly 1`] = `
<div
  className="todo-menu-container"
>
  <input
    className="todo-menu-input"
    onChange={[Function]}
    onClick={[Function]}
    placeholder="Add New Item"
    type="text"
    value=""
  />
  <TodoColorBox
    chosenColor="#eb716a"
    color="#eb716a"
    setColor={[Function]}
  />
  <TodoColorBox
    chosenColor="#eb716a"
    color="#fe74a2"
    setColor={[Function]}
  />
  <TodoColorBox
    chosenColor="#eb716a"
    color="#9170cc"
    setColor={[Function]}
  />
  <TodoColorBox
    chosenColor="#eb716a"
    color="#5cb3f9"
    setColor={[Function]}
  />
  <TodoColorBox
    chosenColor="#eb716a"
    color="#6ed3e5"
    setColor={[Function]}
  />
  <TodoColorBox
    chosenColor="#eb716a"
    color="#ffe07f"
    setColor={[Function]}
  />
  <button
    className="todo-menu-button"
    onClick={[Function]}
  >
    Add
  </button>
</div>
`;
